src/AbstractServer.php

```diff
@@ -139,15 +127,15 @@ protected function _dispatch(Method\Definition $invokable, array $params)
         $callback = $invokable->getCallback();
         $type     = $callback->getType();
 
-        if ('function' == $type) {
+        if ('function' === $type) {
             $function = $callback->getFunction();
             return call_user_func_array($function, $params);
         }
 
         $class  = $callback->getClass();
         $method = $callback->getMethod();
 
-        if ('static' == $type) {
+        if ('static' === $type) {
             return call_user_func_array([$class, $method], $params);
```

Comment:
I think the `Method\Definition` should simply have a `getCallback()` method. We can then use the construct `return $callback(...$params);` everywhere.

Static callbacks can be returned as either `[$class, $method]` or `{$class}::{$method}`; with PHP 7, a variable in either notation is callable, and can be called in the same way as a function. Instance callbacks can be returned as `[$instance, $method]`, and, again, can be called in the same way.

This way, there's no reason to test for `$type` any more. (This was a remnant from the PHP 5 days, when we had to call callbacks in different ways based on how they were constructed.)

Reply:
Thanks for the explanation. I quite like the idea.

Reply:
I gave it some more thought and would like to provide this change in a separate PR, since it is somewhat out of context to strict typing. Would that be alright with you, @weierophinney ?

Reply:
Absolutely.

Reply:
ty